gridData/mrc.py

```diff
@@ -0,0 +1,136 @@
+
+.. _mrcfile: https://mrcfile.readthedocs.io/
+.. _`MRC2014 file format`: http://www.ccpem.ac.uk/mrc_format/mrc2014.php
+
+
+References
+----------
+
+.. [Burnley2017] Burnley T, Palmer C and Winn M (2017) Recent
+                 developments in the CCP-EM software suite. *Acta
+                 Cryst.* D73:469-477. doi: `10.1107/S2059798317007859`_
+
+
+.. _`10.1107/S2059798317007859`: https://doi.org/10.1107/S2059798317007859
+
+Classes
+-------
+
+"""
+from __future__ import absolute_import
+from six.moves import range
+
+import warnings
+
+import numpy as np
+import mrcfile
+
+
+class MRC(object):
+    """Represent a MRC/CCP4 file.
+
+    Load `MRC/CCP4 2014 <MRC2014 file format>`_ 3D volumetric data with
+    the mrcfile_ library.
+
+    Parameters
+    ----------
+    filename : str (optional)
+       input file (or stream), can be compressed
+
+
+    Attributes
+    ----------
+    header : numpy.recarray
+       Header data from the MRC file as a numpy record array.
+
+    array : numpy.ndarray
+       Data as a 3-dimensional array where axis 0 corresponds to X,
+       axis 1 to Y, and axis 2 to Z. This order is always enforced,
+       regardless of the order in the mrc file.
+
+    delta : numpy.ndarray
+       Diagonal matrix with the voxel size in X, Y, and Z direction
+       (taken from the :attr:`mrcfile.mrcfile.voxel_size` attribute)
+
+    origin : numpy.ndarray
+       numpy array with coordinates of the coordinate system origin
+       (taken from :attr:`header.origin`)
+
+    rank : int
+       The integer 3, denoting that only 3D maps are read.
+
+
+    Notes
+    -----
+    * Only volumetric (3D) densities are read.
+    * Only orthorhombic unitcells supported (other raise :exc:`ValueError`)
+    * Only reading is currently supported.
+
+
+    .. versionadded:: 0.7.0
+
+    """
+
+    def __init__(self, filename=None):
+        self.filename = filename
+        if filename is not None:
+            self.read(filename)
+
+    def read(self, filename):
+        """Populate the instance from the MRC/CCP4 file *filename*."""
+        if filename is not None:
+            self.filename = filename
+        with mrcfile.open(filename) as mrc:
+            if not mrc.is_volume():                           #pragma: no cover
+                raise ValueError(
+                    "MRC file {} is not a volumetric density.".format(filename))
+            self.header = h = mrc.header.copy()
+            # check for being orthorhombic
+            if not np.allclose([h.cellb.alpha, h.cellb.beta, h.cellb.gamma],
+                               [90, 90, 90]):
+                raise ValueError("Only orthorhombic unitcells are currently "
+                                 "supported, not "
+                                 "alpha={0}, beta={1}, gamma={2}".format(
+                                     h.cellb.alpha, h.cellb.beta, h.cellb.gamma))
+            # mrc.data[z, y, x] indexed: convert to x,y,z as used in GridDataFormats
```

Comment:
Do we need to do further manipulations here?

```
Note that the MRC format allows the data axes to be swapped using the header’s mapc, mapr and maps fields. This library does not attempt to swap the axes and simply assigns the columns to X, rows to Y and sections to Z. (The data array is indexed in C style, so data values can be accessed using mrc.data[z][y][x].) In general, EM data is written using the default axes, but crystallographic data files might use swapped axes in certain space groups – if this might matter to you, you should check the mapc, mapr and maps fields after opening the file and consider transposing the data array if necessary.
```

Reply:
I agree.  If the file has `mapc, mapr, maps = 3, 2, 1`, then the data should already be in C-order and `np.transpose()` would be unnecessary.  `mapc, mapr, maps = 1, 2, 3` is Fortran-order.  It might be best to check for these two cases and disallow the rest. `np.transpose()` should be able to handle the other cases but it they are probably rare.

Reply:
Good point, this needs to be fixed then. Perhaps that fixes #76, too.

Code welcome!!!

Reply:
Ultimately, this and calculating the origin correctly did fix #76